src/settings.rs

```diff
@@ -207,46 +208,81 @@ pub struct CommonSettings {
     pub diagnostic_endpoint: Option<String>,
 }
 
+#[derive(Deserialize, Clone, Debug, PartialEq)]
```

Comment:
It'd be great to move this into its own module, since it isn't really belonging here. Otherwise lgtm

Reply:
I don't think there's a better place to put it, but open to suggestions.